pkgs/misc/ghostscript/default.nix

```diff
@@ -103,8 +103,11 @@ stdenv.mkDerivation rec {
     done
   '';
 
+  # version string may have 3 elements; dynamic library naming only contains the first 2
+  dylib_version = lib.versions.majorMinor version;
   preFixup = lib.optionalString stdenv.isDarwin ''
-    install_name_tool -change libgs.dylib.${version} $out/lib/libgs.dylib.${version} $out/bin/gs
+    install_name_tool -change libgs.dylib.$dylib_version $out/lib/libgs.dylib.$dylib_version $out/bin/gs
+    $out/bin/gs --version
```

Comment:
This line doesn't belong to preFixup. You could move it to `postCheck` or remove.

Reply:
I understand what you mean!

Unfortunately `postCheck` happens before `preFixup` ... my first attempt was to put it there but it was failing.

I considered not including it, but then there is a silent build failure in the future if linkage issues change. So, in light of that I figured this would be the "lesser evil".

The small consolation is that at least all of this Darwin linkage business is kept in one place, which makes it a bit neater I guess.

Any advice on how to make this cleaner while still maintaining a sanity check? If you feel it's best not to bother I can remove it I suppose.

Reply:
Good point. Consider:
```nix
doInstallCheck = true;
installCheckPhase = ''
  runHook preInstallCheck

  "$out"/bin/gs --version

  runHook postInstallCheck
'';
```

Reply:
Thank you, I appreciate the help.

Updated and force-pushed.